ppa/archive/templatetags/ppa_tags.py

```diff
@@ -113,3 +116,165 @@ def esc(x):
+    return fields
+
+
+def _add_excerpt_fields(item, existing_data):
+    """Return fields specific to excerpts."""
+    fields = {"rft_val_fmt": "info:ofi/fmt:kev:mtx:book", "rft.genre": "bookitem"}
+    if "title" in existing_data:
+        fields["rft.atitle"] = existing_data["title"]
+    if _get_item_value(item, "book_journal"):
+        fields["rft.btitle"] = _get_item_value(item, "book_journal")
+    if _get_item_value(item, "pub_place"):
+        fields["rft.place"] = _get_item_value(item, "pub_place")
+    # Add page information for excerpts if available
+    if _get_item_value(item, "first_page"):
+        first_page = _get_item_value(item, "first_page")
+        last_page = _get_item_value(item, "last_page")
+        fields["rft.spage"] = first_page
+        fields["rft.epage"] = last_page or first_page
+        # For rft.pages, use full range if we have both pages, otherwise just first page
+        if last_page and last_page != first_page:
+            fields["rft.pages"] = f"{first_page}-{last_page}"
+        else:
+            fields["rft.pages"] = first_page
+    return fields
+
+
+def _add_article_fields(item, existing_data):
+    """Return fields specific to articles."""
+    fields = {"rft_val_fmt": "info:ofi/fmt:kev:mtx:journal", "rft.genre": "article"}
+    if "title" in existing_data:
+        fields["rft.atitle"] = existing_data["title"]
+    if _get_item_value(item, "book_journal"):
+        fields["rft.jtitle"] = _get_item_value(item, "book_journal")
+    if _get_item_value(item, "enumcron"):
+        fields["rft.volume"] = _get_item_value(item, "enumcron")
+    if _get_item_value(item, "first_page"):
+        first_page = _get_item_value(item, "first_page")
+        last_page = _get_item_value(item, "last_page")
+        fields["rft.spage"] = first_page
+        fields["rft.epage"] = last_page or first_page
+        # For rft.pages, use full range if we have both pages, otherwise just first page
+        if last_page and last_page != first_page:
+            fields["rft.pages"] = f"{first_page}-{last_page}"
+        else:
+            fields["rft.pages"] = first_page
+    return fields
+
+
+def _add_work_type_specific_fields(item, existing_data, work_type_str):
+    """Return fields based on work type."""
+    if work_type_str == "full-work":
+        return _add_full_work_fields(item, existing_data)
+    elif work_type_str == "excerpt":
+        return _add_excerpt_fields(item, existing_data)
+    elif work_type_str == "article":
+        return _add_article_fields(item, existing_data)
+    return {}
+
+
+def _convert_to_coins_format(existing_data):
+    """Return fields converted to COinS format."""
+    fields = {}
+    if "au" in existing_data:
+        fields["rft.au"] = existing_data["au"]
+    if "date" in existing_data:
+        fields["rft.date"] = existing_data["date"]
+    if "pub" in existing_data:
+        fields["rft.pub"] = existing_data["pub"]
+    return fields
```

Comment:
Do all the data fields get this 'rft' prefix? I like the approach you're taking here and would suggest using it everywhere - use the unprefixed field names everywhere and then add them all at the end (maybe as part of filtering out any nulls).

Maybe something like this:
```suggestion
   return {f"rft.{key}": val for key, val in existing_data.items() if val is not None}
```